### DISCUSSION TITLE
Azuracast Now Playing not showing special characters

### DISCUSSION THREAD

**Discussion by ghost** [AzuraCast/AzuraCast#5786] [Q&A]
Hello. I am having a problem where when there is a special character- an accent or an umlaut, or anything of that nature- the now playing info text will stop at that character, and won't show the rest of the info. The special character and everything after is gone. I am not using AutoDJ, I am using RadioDJ, all looks good there. Encoding to BUTT, all good there. It's when it gets to the Azuracast public page/stream that the issue begins, and the issue persists when using the embed codes on my site. Is there any fix for this along the chain, or am I going to have to go through the collection and fix those letters manually? Thank you!

**Comment by Moonbase59** [comment]
Character set problems following us around forever…

The only advice I have is carefully checking your chain—starting at the FLAC (or MP3) up to the end.

Modern systems use UTF-8 throughout, but for instance in MP3 ID3v1 and ID3v2.3, either ISO-8859-1 or UTF-16 was used. In ID3v2.4, UTF can be used. FLAC (Vorbis Comments) is always UTF-8.

So you should ideally use ID3v2.4 (2nd-best, in my opinion, is ID3v2.3 with UTF-16) for MP3. Then check if your processing chain doesn’t mess up the metadata. Windows software is notorious for that, especially since Windows often uses cp1252 which _looks_ like ISO-8859-1 but _isn’t_. One of the typical problems is MP3s with ID3v2.2 or ID3v2.3 tags being blindly encoded with metadata in a zillion character sets (like, say, Russian) _while they only support ISO-8859-1. In that case, you’d have to repair them, or use something like MusicBrainz Picard as a tagger. Oh, and get rid of any ID3v1 tags in any case—they are much too restricted, and there is no need nowadays to have _both_ ID3v1 and ID3v2 tags in MP3 files.

AzuraCast supports UTF-8 beautifully, as do Icecast and LiquidSoap. BUTT _should_ be okay usually. Don’t know about RadioDJ. Should you build a "real" studio (with mixers etc.), mAirList is a good choice for Windows. If you’re on Linux, you could use IDJC.
